rafthttp/stream.go

```diff
@@ -0,0 +1,279 @@
+package rafthttp
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/coreos/etcd/raft/raftpb"
+)
+
+var (
+	streamBufSize = 4096
+	errHeartbeat  = errors.New("the message is heartbeat")
+)
+
+type encoder interface {
```

Comment:
I think this idea is good. I will try it in this way.
a `encoderFactory` that creates a `encoder` each time the information is updated.